documentation/week3/src/App.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import './App.css';
+import Form from './components/Form';
+import List from './components/List';
+import data from './data/todos.json';
+import { loadFromLocalStorage, saveToLocalStorage } from './storage/localStoarge';
+export default class App extends Component {
+
+  state = loadFromLocalStorage() || {
+    todos: data,
+    id: 1,
+    discription: '',
+    deadline:''
+  }
+
+
+  handleDiscriptionChange = (event) => {
+    event.preventDefault();
+    this.setState({
+      discription: event.target.value
+    })
+  }
+
+  handleDeadlineChange = (event) => {
+    event.preventDefault();
+    this.setState({
+      deadline: event.target.value
+    })
+  }
+
+  handleClick = (id) => {
+    const todos = this.state.todos;
+    todos[id].done = !todos[id].done;
+    this.setState({ todos });
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const newTodo = {
+      discription: this.state.discription,
+      deadline:this.state.deadline,
+      done: false
+    };
+    const todos = this.state.todos;
```

Comment:
The same as the previous comment here, `const todos = this.state.todos;` this doesn't mean you created a **clone** from the todos object but it means you just created another **reference** for it. So when you change the variable **todos**, it will change the original **this.state.todos** because your new variable is just a **reference** .

So that, `todos.push(newTodo)` will push this `newTodo` to the **this.state.todos**.

To make a clone from an array you can do like this `const todos = [...this.state.todos];`